packages/wrangler/src/__tests__/workflows.test.ts

```diff
@@ -780,5 +780,30 @@ describe("wrangler workflows", () => {
 			await expect(runWrangler("deploy --dry-run")).rejects.toThrow();
 			expect(std.err).toContain('"workflows" bindings should be objects');
 		});
+
+		it("should reject workflows binding with same name", async () => {
+			writeWorkerSource({ format: "ts" });
+			writeWranglerConfig({
+				main: "index.ts",
+				workflows: [
+					{
+						binding: "MY_WORKFLOW",
+						name: "valid-workflow-name",
+						class_name: "MyWorkflow",
+						script_name: "external-script",
+					},
+					{
+						binding: "MY_WORKFLOW_2",
+						name: "valid-workflow-name",
+						class_name: "MyWorkflow2",
+						script_name: "external-script-2",
+					},
+				],
+			});
+			await expect(runWrangler("deploy --dry-run")).rejects.toThrow();
+			expect(std.err).toContain(
+				'"workflows" bindings must have unique "name" values'
+			);
```

Comment:
Consider using an inline snapshot here so that we can also get a check on the diagnostic message (e.g. which names are duplicated):

```suggestion
			expect(std.err).toMatchInlineSnapshot();
```

When you next run the test it will write the snapshot into this call as a parameter.